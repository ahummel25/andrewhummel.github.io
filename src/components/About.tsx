import React, { forwardRef, FC, RefObject, Ref } from "react";
import styled from "@emotion/styled";
import { dimensions, colors, fonts } from "../styles/variables";

const StyledAboutContainer = styled.div`
  height: 100%;
  width: 100%;
  background-color: ${colors.white};
  border: 0;
  display: flex;
  justify-content: center;
`;

const StyledAboutHeader = styled.h2`
  font-family: ${fonts.helvetica};
  font-size: ${dimensions.fontSize.profileText}px;
  font-weight: ${dimensions.fontWeight.bold};
  color: ${colors.profileText};
  margin: 1.5rem 0;
  text-transform: uppercase;
`;

const StyledHr = styled.hr`
  position: relative;
  color: ${colors.profileText};
  margin: 0 auto;
  max-width: 250px;
  border: 0;
  border-top: 4px solid;
  bottom: 15px;
`;

const StyledAboutBodyLeft = styled.div`
  width: 50%;
  float: left;
`;

const StyledAboutBodyRight = styled.div`
  width: 50%;
  float: right;
`;

const StyledAboutText = styled.p`
  font-family: ${fonts.helvetica};
  font-size: ${dimensions.fontSize.large + 2}px;
  color: ${colors.profileText};
`;

const About: FC<{}> = (_, ref: RefObject<HTMLDivElement>): JSX.Element => (
  <div ref={ref}>
    <StyledAboutContainer>
      <StyledAboutHeader>About Me</StyledAboutHeader>
    </StyledAboutContainer>
    <StyledHr />
    <StyledAboutBodyLeft>
      <StyledAboutText>
        I have over 7 years of experience as a full-stack software engineer and
        have worked at both small startups and large organizations. While I'm a
        proficient full-stack developer, my expertise is in building scalable
        backend services (API services, stream processing, and async
        mechanisms).
      </StyledAboutText>
    </StyledAboutBodyLeft>
    <StyledAboutBodyRight>
      <StyledAboutText>
        I have an insatiable interest in learning new technologies as well as a
        strong desire to build modular, maintainable, and well-tested
        applications.
      </StyledAboutText>
    </StyledAboutBodyRight>
  </div>
);

export default forwardRef<HTMLDivElement>(About);
