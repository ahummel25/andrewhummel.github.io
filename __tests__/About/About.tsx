import React from 'react';
import renderer from 'react-test-renderer';

import About from '../../src/components/About';

describe('Header', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<About />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
