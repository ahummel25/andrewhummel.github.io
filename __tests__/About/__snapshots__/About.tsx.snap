// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
.emotion-3 {
  margin-bottom: 20px;
  padding-top: 0;
}

.emotion-0 {
  margin: 0 auto;
  background-color: rgb(0,128,115);
  font-weight: 100;
  padding: 30px 40px;
  margin-bottom: 60px;
  text-transform: uppercase;
}

@media (max-width:992px) {
  .emotion-0 {
    padding: 20px 30px;
    margin-bottom: 40px;
  }
}

.emotion-2 {
  width: 95%;
  font-family: Roboto,sans-serif;
  font-weight: 300;
  padding: 0 24px;
}

.emotion-1 {
  border-left: 5px solid rgb(121,85,72);
  color: rgb(121,85,72);
  font-weight: 400;
  font-size: 125%;
  margin: 20px 0;
  padding-left: 24px;
}

.emotion-1 h2 {
  font-size: inherit;
  line-height: inherit;
}

.emotion-3 {
  margin-bottom: 20px;
  padding-top: 0;
}

.emotion-0 {
  margin: 0 auto;
  background-color: rgb(0,128,115);
  font-weight: 100;
  padding: 30px 40px;
  margin-bottom: 60px;
  text-transform: uppercase;
}

@media (max-width:992px) {
  .emotion-0 {
    padding: 20px 30px;
    margin-bottom: 40px;
  }
}

.emotion-2 {
  width: 95%;
  font-family: Roboto,sans-serif;
  font-weight: 300;
  padding: 0 24px;
}

.emotion-1 {
  border-left: 5px solid rgb(121,85,72);
  color: rgb(121,85,72);
  font-weight: 400;
  font-size: 125%;
  margin: 20px 0;
  padding-left: 24px;
}

.emotion-1 h2 {
  font-size: inherit;
  line-height: inherit;
}

<section
  className="section scrollspy emotion-3"
  id="about"
>
  <h3
    className="page-title white-text emotion-0"
  >
    About
  </h3>
  <div
    className="container flow-text emotion-2"
  >
    <blockquote
      className="emotion-1"
    >
      <h2>
        Highly motivated, full stack software developer with a passion for delivering high quality, requirements-driven software.
      </h2>
    </blockquote>
    <p>
      I have over 7 years of experience as a full stack software engineer and have worked at both small startups and large organizations. As a software developer, I enjoy putting my obsessive attention to detail to use building out user friendly frontend components and scalable backend services.
    </p>
    <p>
      I have an insatiable interest in learning new technologies as well as a strong desire to build modular, maintainable, and well-tested applications. I am constantly sharpening my skills and keeping up with the latest trends in web development and cloud computing.
    </p>
  </div>
</section>
`;
